### DISCUSSION TITLE
named vectors

### DISCUSSION THREAD

**Discussion by frutik** [qdrant/qdrant#5044] [Show and tell]
from the docs 

> Named Vectors
Aside from storing multiple vectors of the same shape in a single point, Qdrant supports storing multiple different vectors in a single point.

> Each of these vectors should have a unique configuration and should be addressed by a unique name. 

and no example neither hot to insert the data nor how to query it.

i am trying to bruteforce syntax of search and get

> {
  "error": "Format error in JSON body: data did not match any variant of untagged enum NamedVectorStruct at line 1 column 10410"
}

name of vector is resnet

query is

`POST collections/adcurve/points/search
{
  "with_payload": true,
  "vector": {
    "resnet": []}}`

**Comment by Anush008** [comment]
Hey @frutik. You can refer to the following sections in the docs:
- [Adding points with named vectors](https://qdrant.tech/documentation/concepts/points/#upload-points:~:text=If%20the%20collection%20was%20created%20with%20multiple%20vectors%2C%20each%20vector%20data%20can%20be%20provided%20using%20the%20vector%E2%80%99s%20name%3A)
- [Searching with named vectors](https://qdrant.tech/documentation/concepts/search/#search-api:~:text=If%20the%20collection%20was%20created%20with%20multiple%20vectors%2C%20the%20name%20of%20the%20vector%20to%20use%20for%20searching%20should%20be%20provided%3A)

We are working on a new section to bring these examples together.

**Reply by Anush008** [reply]
The new section is at https://qdrant.tech/documentation/concepts/vectors/#named-vectors

**Comment by coszio** [comment]
Complementing the docs, we also have this detailed API reference you can use for these kind of specifics :)

### Solution

In this case [the reference indicates](https://api.qdrant.tech/api-reference/search/points#request.body.vector) that named vector mode for `search` should look like this:

```
POST collections/adcurve/points/search 
{ 
  "with_payload": true, 
  "vector": { 
    "vector": [<your query vector>], 
    "name": "resnet"
  }
}
```

### Alternative

In the superseding [Query API](https://api.qdrant.tech/api-reference/search/query-points), we solved this by introducing a `"using"` parameter, which you can set up like this:

```
POST collections/adcurve/points/query 
{ 
  "query": [<your query vector>],
  "using": "resnet",
  "with_payload": true
}
```

**Comment by frutik** [comment]
Thank you. It works
